test/test_container_example_apps.py

```diff
@@ -0,0 +1,176 @@
+        )
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", expected_output="NGINX is working"
+        )
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", expected_output="NGINX2 is working",
+            host="localhost2"
+        )
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", expected_code=404,
+            page="/nginx-cfg/default.conf"
+        )
+
+
+class TestNginxExamplePerlAppContainer:
+
+    def setup_method(self):
+        self.s2i_app = build_s2i_app(perl_test_app)
+
+    def teardown_method(self):
+        self.s2i_app.cleanup()
+
+    def test_run_app_test(self):
+        if VARS.VERSION.endswith("-micro"):
+            pytest.skip("Run the chosen tests (not for micro variant which lacks perl)")
+        cid_file_name = self.s2i_app.app_name
+        self.s2i_app.set_new_image(image_name=f"{VARS.IMAGE_NAME}-{cid_file_name}")
+        assert self.s2i_app.create_container(cid_file_name=cid_file_name, container_args="--user 10001")
+        cid = self.s2i_app.get_cid(cid_file_name=cid_file_name)
+        assert cid
+        cip = self.s2i_app.get_cip(cid_file_name=cid_file_name)
+        assert cip
+        # Checks if returns header of Perl version
+        perl_version = PodmanCLIWrapper.podman_exec_shell_command(cid_file_name=cid, cmd="perl -e 'print \"$^V\"'")
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", port=8080,
+            expected_output=f"X-Perl-Version: {perl_version}"
+        )
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", port=8080, page="/perl", expected_output="Perl location handler is working"
+        )
+
+
+class TestNginxLogContainer:
+
+    def setup_method(self):
+        self.s2i_app = build_s2i_app(test_app)
+
+    def teardown_method(self):
+        self.s2i_app.cleanup()
+
+    def test_log_output(self):
+        """
+        Test checks if logging container works properly
+        and logs contain proper output
+        """
+        cid_file_name = "test-app"
+        self.s2i_app.set_new_image(image_name=f"{VARS.IMAGE_NAME}-{cid_file_name}")
+        assert self.s2i_app.create_container(
+            cid_file_name=cid_file_name,
+            container_args="--user 10001"
+        )
+        cid = self.s2i_app.get_cid(cid_file_name=cid_file_name)
+        assert cid
+        cip = self.s2i_app.get_cip(cid_file_name=cid_file_name)
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", port=8080, expected_output="NGINX is working"
+        )
+        assert '"GET / HTTP/1.1" 200' in self.s2i_app.get_logs(cid_file_name=cid_file_name)
+        # Check if /nothing-at-all is really no accessible
+        assert self.s2i_app.test_response(
+            url=f"http://{cip}", port=8080, page="/nothing-at-all", expected_code=404
+        )
+        logs = self.s2i_app.get_logs(cid_file_name=cid_file_name)
+        assert logs
```

Comment:
This is kinda duplicate. If `logs` is empty, the next assert fails because it won't find anything in an empty string.